src/Events/Category_Colors/Event_Category_Meta.php

```diff
@@ -0,0 +1,268 @@
+			throw new InvalidArgumentException( __( 'Invalid term ID.', 'the-events-calendar' ) );
+		}
+
+		$term = get_term( $term_id, $this->taxonomy );
+
+		if ( ! $term instanceof WP_Term || is_wp_error( $term ) ) {
+			throw new InvalidArgumentException(
+			/* translators: %1$d is the term ID, %2$s is the taxonomy name. */
+				sprintf( __( 'Term ID %1$d does not exist in taxonomy %2$s.', 'the-events-calendar' ), $term_id, $this->taxonomy )
+			);
+		}
+
+		$this->term_id = $term_id;
+	}
+
+	/**
+	 * Sets metadata for the term.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key   The meta key to update.
+	 * @param mixed  $value The value to store.
+	 *
+	 * @return $this Returns the instance for chaining.
+	 */
+	public function set( string $key, $value ): self {
+		$key   = $this->validate_key( $key );
+		$value = $this->validate_value( $value );
+
+		// If key or value is invalid, skip but don't break chaining.
+		if ( is_wp_error( $key ) || is_wp_error( $value ) ) {
+			return $this;
+		}
+
+		$this->pending_updates[ $key ] = $value;
+
+		return $this;
+	}
+
+	/**
+	 * Marks metadata for deletion but does not delete immediately.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key The meta key to delete.
+	 *
+	 * @return $this|wp_error  Returns the instance for chaining.
+	 */
+	public function delete( string $key ): self {
+		$key = $this->validate_key( $key );
+
+		if ( is_wp_error( $key ) ) {
+			return $key;
+		}
+
+		$this->pending_deletes[] = $key;
+
+		return $this;
+	}
+
+	/**
+	 * Save all queued meta updates and deletions.
+	 *
+	 * @since TBD
+	 *
+	 * @return $this
+	 */
+	public function save(): self {
+		foreach ( $this->pending_updates as $key => $value ) {
+			update_term_meta( $this->term_id, $key, $value );
+		}
+
+		foreach ( $this->pending_deletes as $key ) {
+			delete_term_meta( $this->term_id, $key );
+		}
+
+		// Clear queues after saving.
+		$this->pending_updates = [];
+		$this->pending_deletes = [];
+
+		return $this;
+	}
+
+	/**
+	 * Retrieves metadata for the term.
+	 *
+	 * @since TBD
+	 *
+	 * @param string|null $key Optional. The meta key to retrieve.
+	 *
+	 * @return mixed The meta value, or an array of all metadata if no key is provided.
+	 */
+	public function get( ?string $key = null ) {
+		if ( null === $key ) {
+			return get_term_meta( $this->term_id );
+		}
+
+		$key = $this->validate_key( $key );
+
+		// Check if the key exists before retrieving the value.
+		if ( ! metadata_exists( 'term', $this->term_id, $key ) ) {
+			return null;
+		}
+
+		$value = get_term_meta( $this->term_id, $key, true );
+
+		// Ensure numbers are always returned as strings.
+		if ( is_numeric( $value ) ) {
+			return (string) $value;
+		}
+
+		// Ensure arrays remain unchanged.
+		if ( is_array( $value ) ) {
+			return $value;
+		}
+
+		// Ensure explicitly stored `false` returns an empty string.
+		if ( false === $value ) {
+			return '';
+		}
+
+		// Ensure empty strings remain empty.
+		return ( '' === $value ) ? '' : $value;
```

Comment:
I feel like this can be condensed. Casting `false` and `null` to a string actually results in an empty string, so we can use that to our advantage:
```suggestion
		// Ensure arrays remain unchanged.
		if ( is_array( $value ) ) {
			return $value;
		}

		// Ensure we return a string.
		return (string) $value;
```